xbmc/pvr/addons/PVRClients.cpp

```diff
@@ -176,6 +178,7 @@ void CPVRClients::Unload(void)
   m_strPlayingClientName = "";
 
   m_clientMap.clear();
+  m_loadedEvent.Reset();
```

Comment:
You have instantiated an automatic event. Means that wait resets it. You should reset it when starting the thread here and signal here on exit. IMO you can reduce the timeout as well. It just spins another loop if it times out, right?